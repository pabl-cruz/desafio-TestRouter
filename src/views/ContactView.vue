<script></script>
<template>
<h1>Contacto</h1>
<form>
<input type="text" name="nombre" placeholder="ingrese nombre">
<input type="email" name="correo" placeholder="ingrese correo">
<input type="submit" name="enviar" value="Enviar">
</form>
</template>