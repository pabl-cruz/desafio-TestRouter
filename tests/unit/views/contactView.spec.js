import { mount } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'

import ContactView from '@/views/ContactView.vue'



describe('ContactView', () => { 
    
    it('captura de contenido HTML en prueba y contrasta', () => {
        const wrapper = mount(ContactView)
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('Probando la existencia de vista ContactView ', async () => {
        const router = createRouter({
            history: createWebHistory(),
            routes: [{
                path: '/contact',
                name: 'ContactView',
                component: ContactView
            }],
        })
        router.push('/contact')
        await router.isReady()

        const wrapper = mount(ContactView, {
            global: {
                plugins: [router]
            }
        })
        expect(wrapper.findComponent(ContactView).exists()).toBe(true)
    })  

    test('EstÃ¡ presente formulario', async ()=> {
        const wrapper = mount(ContactView)
        const form = wrapper.find('form')
        expect(form.exists()).toBe(true)
    })
 })