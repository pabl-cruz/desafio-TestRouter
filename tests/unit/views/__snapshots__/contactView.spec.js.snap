// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactView captura de contenido HTML en prueba y contrasta 1`] = `
<h1>Contacto</h1>
<form><input type="text" name="nombre" placeholder="ingrese nombre"><input type="email" name="correo" placeholder="ingrese correo"><input type="submit" name="enviar" value="Enviar"></form>
`;
